Lib/functools.py

```diff
@@ -326,6 +326,10 @@ def _partial_new(cls, func, /, *args, **keywords):
                             "or a descriptor")
     if args and args[-1] is Placeholder:
         raise TypeError("trailing Placeholders are not allowed")
+    if keywords:
+        for v in keywords.values():
+            if v is Placeholder:
+                raise TypeError("keyword Placeholders are not allowed")
```

Comment:
```suggestion
    if Placeholder in keyword.values():
        raise TypeError("keyword Placeholders are not allowed")
```

Reply:
Will leave conditional for performance, but yeah, that is better than loop.

Reply:
> Will leave conditional for performance, but yeah, that is better than loop.

Please write the code pythonic and easy to read. _Iff_ `d and thing in d` is faster than `thing in d`, we need to improve the speed of the latter, _not_ promote the more complicated former. Please consider again my suggestion.

Reply:
```py
$ python3.14 -m timeit -s "d = {}" "'a' in d"
50000000 loops, best of 5: 10.1 nsec per loop
$ python3.14 -m timeit -s "d = {}" "d and 'a' in d"
20000000 loops, best of 5: 17.8 nsec per loop
```

Reply:
For `.values()`:

```py
$ python3.14 -m timeit -s "d = {}" "d and 'a' in d.values()"
20000000 loops, best of 5: 17.7 nsec per loop
$ python3.14 -m timeit -s "d = {}" "'a' in d.values()"
5000000 loops, best of 5: 52.2 nsec per loop
```

Reply:
It is better on your machine than on mine.

OSX 3.14:
```
$PYMAIN -m timeit -s "d = {}" "d and 'a' in d.values()"
20000000 loops, best of 5: 17.8 nsec per loop
$PYMAIN -m timeit -s "d = {}" "'a' in d.values()"
2000000 loops, best of 5: 99.6 nsec per loop
```

Reply:
Make sure it's an optimised build.

Reply:
It is.

So ok, let's say difference is in the range of 3x - 5x. Of course it is worth noting that in case that `dict` is not empty it adds a slight overhead. But in this case, I would say bigger portion of calls are done without keywords.
The pattern itself is not too bad.
The reason is different, but somewhat reminds: `if list and list[-1] is None:`.

What is best here?

Reply:
I guess it's fine as it is now.